Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Ballistic.cs

```diff
@@ -71,18 +77,63 @@ private void OnBallisticAfterInteract(EntityUid uid, BallisticAmmoProviderCompon
             Deleted(args.Target) ||
             !TryComp<BallisticAmmoProviderComponent>(args.Target, out var targetComponent) ||
             targetComponent.Whitelist == null)
-        {
             return;
-        }
 
         args.Handled = true;
 
-        // Continuous loading
-        _doAfter.TryStartDoAfter(new DoAfterArgs(EntityManager, args.User, component.FillDelay, new AmmoFillDoAfterEvent(), used: uid, target: args.Target, eventTarget: uid)
+        ReloadDoAfter(component, args.User, args.Target.Value, args.Used);
+    }
+
+    private void OnBallisticAfterInteractUsing(EntityUid uid, BallisticAmmoProviderComponent component, AfterInteractUsingEvent args)
+    {
+        if (args.Handled ||
+            !Timing.IsFirstTimePredicted ||
+            args.Target == null ||
+            args.Used == args.Target ||
+            Deleted(args.Target) ||
+            !TryComp<BallisticAmmoProviderComponent>(args.Target, out var targetComponent) ||
+            targetComponent.ReloadFromClothing == false)
+            return;
+
+        args.Handled = true;
+
+        if (!TryComp<ClothingComponent>(args.Target, out var clothing)
+            || clothing.InSlot == null
+            || clothing.InSlotFlag == SlotFlags.POCKET)
+            return;
+
+        ReloadDoAfter(component, args.User, args.Used, args.Target.Value);
+    }
+
+    private void OnSmartEquipWithItemAttempt(Entity<BallisticAmmoProviderComponent> ent, ref SmartEquipWithItemAttemptEvent args)
+    {
+        if (!Timing.IsFirstTimePredicted
+            || ent.Comp.ReloadFromClothing == false
+            || !TryComp<ClothingComponent>(args.SlotEntity, out var clothing)
```

Comment:
```suggestion
            || !TryComp<ClothingComponent>(ent.Owner, out var clothing)
```

Gonna be real, I don't know why it supplies `SlotEntity` if the event is raised on the entity anyways.

Reply:
Removed!